### PR TITLE
ci: Build scanner packages natively

### PR DESCRIPTION
**Pull request by dividedmind**
@apotterri @kgilpin I just replaced appmap with scanner where it makes sense.

@dustinbyrne note it need the same fix for npm tag as the other workflow.

### CODE REVIEW

**Comment by dividedmind** [comment]
Note, I noticed, also needs changing `.releaserc.js` to not auto-release scanner.

File: .github/workflows/build-native-scanner.yml [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+      - name: Sign the release with Trusted Signing
+        uses: azure/trusted-signing-action@v0.3.16
+        with:
+          azure-tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          azure-client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          azure-client-secret: ${{ secrets.AZURE_CLIENT_SECRET }}
+          endpoint: https://eus.codesigning.azure.net/
+          code-signing-account-name: appmap
+          certificate-profile-name: appmap
+          files-folder: ${{ github.workspace }}/packages/cli/release/
```
Comment (dustinbyrne) [inline .github/workflows/build-native-scanner.yml:149]: This path looks like it should change

File: .github/workflows/build-native-scanner.yml [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+          yarn build
+          cd packages/scanner
+          yarn build-native linux x64
+          ${GITHUB_WORKSPACE}/bin/hash release/scanner-linux-x64
+
+      - name: Publish scanner-linux-x64 artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: scanner-linux-x64
+          path: packages/cli/release/scanner-linux-x64
```
Comment (dustinbyrne) [inline .github/workflows/build-native-scanner.yml:32]: packages/scanner

File: .github/workflows/build-native-scanner.yml [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+        run: |
+          python3 -m pip install --break-system-packages setuptools
+
+      - name: Setup Node
+        uses: ./.github/actions/setup-node
+
+      - name: Build
+        run: |
+          yarn build
+          cd packages/cli
```
Comment (dustinbyrne) [inline .github/workflows/build-native-scanner.yml:57]: packages/scanner

File: .github/workflows/build-native-scanner.yml [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+          yarn build-native macos x64
```
Comment (dustinbyrne) [inline .github/workflows/build-native-scanner.yml:58]: I had to update the `build-native` script to take a platform/arch in this manner.

Comment (dividedmind) [reply .github/workflows/build-native-scanner.yml:58]: what do you mean? I copied this from build-native.yml

Comment (dividedmind) [reply .github/workflows/build-native-scanner.yml:58]: ah, ok, I get it

File: packages/scanner/bin/build-native [hunk 5]
```diff
@@ -1,9 +1,21 @@
...
+
+if [ -n "${OS}" ]; then
+  ARCH="${2:-x64}"
+  echo "Building native binaries for ${OS} ${ARCH}..."
+  ARGS="${ARGS} -o release/scanner-${OS}-${ARCH} -t node18-${OS}-${ARCH}"
+else
+  echo "Building native binaries for each target..."
+fi
+
+yarn pkg ${ARGS} built/cli.js
```
[the remark this answers is not in the document]
Comment (dustinbyrne) [reply packages/scanner/bin/build-native:21]: Actually this looks correct

[the remark this answers is not in the document]
Comment (dividedmind) [reply packages/scanner/bin/build-native:21]: it is

**Pull request merged**